### PR TITLE
DATAMONGO-1608 - Fixed potential NullPointerException for derived ignore case query methods

### PR DESCRIPTION
**Pull request by EdwardPrentice**
- [x] You have read the [Spring Data contribution guidelines](https://github.com/spring-projects/spring-data-build/blob/master/CONTRIBUTING.adoc).
- [x] There is a ticket in the bug tracker for the project in our [JIRA](https://jira.spring.io/browse/DATAMONGO).
- [x] You use the code formatters provided [here](https://github.com/spring-projects/spring-data-build/tree/master/etc/ide) and have them applied to your changes. Don’t submit any formatting related changes.
- [x] You submit test cases (unit or integration tests) that back your changes.
- [x] You added yourself as author in the headers of the classes you touched. Amend the date range in the Apache license header if needed. For new types, add the license header (copy from another file and set the current year only).

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@EdwardPrentice Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-data-mongodb&pullRequestId=439)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-projects/spring-data-mongodb&pullRequestId=439) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@EdwardPrentice Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-data-mongodb&pullRequestId=439)!

File: spring-data-mongodb/src/main/java/org/springframework/data/mongodb/repository/query/MongoQueryCreator.java [hunk 1]
```diff
@@ -15,6 +15,7 @@
  */
 package org.springframework.data.mongodb.repository.query;
 
+import static java.util.Objects.nonNull;
```
Comment (olivergierke) [inline spring-data-mongodb/src/main/java/org/springframework/data/mongodb/repository/query/MongoQueryCreator.java]: We need to preserve JDK 6 compatibility. Can we just exchange this against a simple null check?

**Comment by EdwardPrentice** [comment]
Made change to remove nonNull for JDK 6 compatibility. Tests don't all pass because I can't work out what should happen in the null case - all implementations I've come up with so far cause a different number of other tests to fail.

**Comment by EdwardPrentice** [comment]
Done :) At least works, and passes all tests.

**Comment by christophstrobl** [comment]
Thanks @EdwardPrentice! Applied to master and ported to SR branches.

**Pull request closed without merging**